### PR TITLE
fix #5234, #5255: better report state handling

### LINKED ISSUE #5234: 20221215 react update plots as simulation is running
- clear plots when "start new simulation" is selected
- when frames are available from runStatus, update the plots rather than waiting for the full simulation to finish

### PR DESCRIPTION
**Pull request by garsuga**
also fixes:
- reports will 'carry on' polling and showing status after refresh
- heatplots do their min/max math correctly now
- reports will seek latest frame on refresh
- theres a WIP type for the status returned by requests like run-status
- additional states can be handled by the start simulation layout like missing and error

### CODE REVIEW

File: react/src/data/report.ts [hunk 1]
```diff
@@ -1,49 +1,96 @@
...
+        report: string
     }) => {
-        runStatus({
+        pollRunReport({
             appName,
             models,
             simulationId,
             report,
-            forceRun: false,
+            pollInterval: 500,
```
Comment (robnagler) [inline react/src/data/report.ts:90]: This comes from the server.

File: react/src/data/report.ts [hunk 2]
```diff
@@ -1,49 +1,96 @@
...
+     * Called once when reports are starting to be generated
+     */
+    onStart?: () => void;
+    /**
+     * Called once when reports are finished being generated
+     */
+    onComplete?: () => void;
+}
+
+type RunStatusParams = {
```
Comment (robnagler) [inline react/src/data/report.ts:24]: These params are returned from the server in nextRequest from runSimulation and runStatus. They should be treated opaquely by the client. nextRequestSeconds is also returned, and that should be used for the poll on the next polling. Different simulations return different amounts.

Please look at the [angular code](https://github.com/radiasoft/sirepo/blob/8fb85dc7befa7684ac4de4631b053c75d47a014a/sirepo/package_data/static/js/sirepo.js#L2558) to see how to handle requests from the server.

**Review by robnagler — DISMISSED** [review]

File: react/src/data/report.ts (hunk 1 above)
Comment (garsuga) [reply react/src/data/report.ts]: fixed in latest

File: react/src/data/report.ts (hunk 2 above)
Comment (garsuga) [reply react/src/data/report.ts:24]: In order to "pick up where we left off" on reports during a page refresh these are sometimes manually provided. The client should also probably know about these so we know which requests can share parameters (like when running a report also begins polling).

Comment (robnagler) [reply react/src/data/report.ts:24]: This isn't how the current app works. It simply calls runSimulation with all the data and the server decides if the simulation needs to run. This same logic could be used.

**Comment by moellep** [comment]
Works much better - here are a few things I noticed with JSPEC:

- When the manual report is "running" but is at 0% complete, show the progress bar at 100% and animated. Some codes take many minutes to set up before data is returned and looking at a empty progress bar may look like it isn't doing anything.
- In the JSPEC "DC Cooling Example", only the IBS & Cooling Rates panel is showing partial updates. The RMS Ion Beam Evolution report should also be showing partial results. Set the Total Number of Steps and Total SImulation Time to 500 to run the report across a longer time period to test.
- In the JSPEC "Booster Ring" example, the IBS & Cooling Rates is showing the next/prev animation buttons, but that report does not have frames.

**Comment by garsuga** [comment]
@moellep I set the default to 100% when there has not been a successful status and tried out a minimum percentage at 5% for the bar to aid with looking empty. Would you rather have it set so 0 shows as 100% or have a minimum percent?

File: react/src/data/report.ts (hunk 2 above)
Comment (garsuga) [reply react/src/data/report.ts:24]: Would this be a case where forceRun=false?

**Comment by moellep** [comment]
Yes, 5% minimum would be fine, and also non 0% would also be OK.

File: react/src/data/report.ts (hunk 2 above)
Comment (robnagler) [reply react/src/data/report.ts:24]: Correct. The forceRun=True is for manual starts.

Comment (moellep) [reply react/src/data/report.ts:24]: It looks like the nextRequestSeconds has been addressed, so I'll resolve this item.

**Pull request merged**